src/ADP.Portal.Core/Git/Services/GitHubService.cs

```diff
@@ -0,0 +1,210 @@
+    {
+        if (IsUnchanged(currentTeam.Description, team.Description)
+            && IsUnchanged(currentTeam.IsPublic, team.IsPublic)
+            && IsUnchanged(currentTeam.Name, team.Name))
+        {
+            update = null;
+            return false;
+        }
+
+        update = new(team.Name)
+        {
+            Description = team.Description ?? currentTeam.Description,
+            Privacy = (team.IsPublic ?? currentTeam.IsPublic) ? TeamPrivacy.Closed : TeamPrivacy.Secret
+        };
+        return true;
+    }
+
+    private static bool IsUnchanged<T>(T current, T? update)
+    {
+        return update is null || Equals(current, update);
+    }
+
+    private static Dictionary<string, TeamRole> BuildTeamRoleDictionary(IEnumerable<string>? maintainers, IEnumerable<string>? members)
+    {
+        maintainers ??= [];
+        members ??= [];
+
+        members = members.Except(maintainers);
+
+        return Enumerable.Concat(
+            maintainers.Select(m => KeyValuePair.Create(m, TeamRole.Maintainer)),
+            members.Select(m => KeyValuePair.Create(m, TeamRole.Member)))
+            .ToDictionary();
+    }
+
+    private async Task<GithubTeamDetails> CreateTeamAsync(GithubTeamUpdate team)
+    {
+        var request = new NewTeam(team.Name)
+        {
+            Description = team.Description,
+            Privacy = (team.IsPublic ?? true) ? TeamPrivacy.Closed : TeamPrivacy.Secret,
+        };
+
+        foreach (var member in team.Maintainers ?? [])
+            request.Maintainers.Add(member);
+
+        logger.LogInformation("Creating team {TeamName}.", team.Name);
+        var newTeam = await client.Organization.Team.Create(options.Value.Organisation, request);
+
+        logger.LogInformation("Team {TeamName} ({TeamId}) has been created, syncing members.", newTeam.Name, newTeam.Id);
+        await SyncTeamMembers(newTeam.Id, [], BuildTeamRoleDictionary(null, team.Members));
+
+        return new()
+        {
+            Id = newTeam.Id,
+            Name = newTeam.Name,
+            Slug = newTeam.Slug,
+            Description = newTeam.Description,
+            IsPublic = newTeam.Privacy.Value == TeamPrivacy.Closed,
+            Maintainers = request.Maintainers,
+            Members = team.Members ?? []
+        };
+    }
+
+    private async Task SyncTeamMembers(int teamId, Dictionary<string, TeamRole> currentMembers, Dictionary<string, TeamRole> targetMembers)
+    {
+        var setMembers = targetMembers
+            .Where(kvp => !currentMembers.TryGetValue(kvp.Key, out var currentRole) || currentRole < kvp.Value)
+            .Select(m => SetMemberRole(m.Key, m.Value));
+        var removeMembers = currentMembers.Keys.Except(targetMembers.Keys)
+            .Select(RemoveMember);
+
+        foreach (var task in setMembers.Concat(removeMembers))
+            await task; // Can we do this in parallel maybe? Doing it like this might be better for the ratelimiting, but will be slower.
```

Comment:
We probably won't be processing many members at a time, I would go with the more basic implementation even though it will be slower.

Reply:
@ppotturi  @ajaysoni-cap  @DaRoszko  what do you think on the above implementation?

Reply:
reducing calls is OK for now.